RUFAS/biophysical/animal/nutrients/nutrition_evaluator.py

```diff
@@ -0,0 +1,317 @@
+
+
+class NutritionEvaluator:
+    """Checks if energy and nutrients supplied in a ration satisfy the demand for an individual animal."""
+
+    @classmethod
+    def evaluate_nutrition_supply(
+        cls,
+        requirements: NutritionRequirements,
+        supply: NutritionSupply,
+        is_cow: bool,
+    ) -> tuple[bool, NutritionEvaluationResults]:
+        """
+        Calculates the difference between nutrient demand and supply, if any.
+
+        Parameters
+        ----------
+        requirements : NutritionRequirements
+            Energy and nutrition requirements of an animal against which the nutrient supply will be compared.
+        supply : NutritionSupply
+            Energy and nutrition supply against which an animal's nutrient requirements will be compared.
+        is_cow : bool
+            True if the animal consuming the ration is a cow, false if it is a heifer.
+
+        Returns
+        -------
+        tuple[bool, NutritionEvaluationResults]
+            Boolean indicating if the ration has an amount of energy and nutrients sufficient to meet the given
+            requirements, and an object containing a summary of all energy and nutrient surpluses and deficiencies.
+
+        """
+        heifer_energy_nutrition_checkers = {
+            "maintenance": cls._check_activity_maintenance_energy_supplied,
+            "growth": cls._check_growth_energy_supplied,
+            "calcium": cls._check_calcium_supplied,
+            "phosphorus": cls._check_phosphorus_supplied,
+            "protein": cls._check_protein_supplied,
+            "ndf": cls._check_neutral_detergent_fiber_supplied,
+            "fat": cls._check_fat_content,
+            "dry_matter": cls._check_dry_matter_intake,
+        }
+        cow_energy_nutrition_checkers = heifer_energy_nutrition_checkers + {
+            "total_energy": cls._check_total_energy_supplied,
+            "lactation": cls._check_lactation_energy_supplied,
+        }
+
+        checkers = cow_energy_nutrition_checkers if is_cow else heifer_energy_nutrition_checkers
+        results = {name: method(requirements, supply) for name, method in checkers.items()}
+
+        evaluation = NutritionEvaluationResults(
+            total_energy=results.get("total_energy"),
+            maintenance_energy=results["maintentance"],
+            lactation_energy=results.get("lactation"),
+            growth_energy=results["growth"],
+            calcium=results["calcium"],
+            phosphorus=results["phosphorus"],
+            metabolizable_protein=results["protein"],
+            ndf_percent=results["ndf"],
+            fat_percent=results["fat"],
+            dry_matter=results["dry_matter"],
+        )
+
+        is_valid_ration = evaluation.is_valid_cow_ration if is_cow else evaluation.is_valid_heifer_ration
+        return is_valid_ration, evaluation
+
+    @classmethod
+    def _check_total_energy_supplied(requirements: NutritionRequirements, supply: NutritionSupply) -> float:
```

Comment:
```suggestion
    def _calculate_total_energy_supplied(requirements: NutritionRequirements, supply: NutritionSupply) -> float:
```
Seems more like a calculation than a check (and matches the docstring definition better which I think is an accurate description).